gammapy/irf/energy_dispersion.py

```diff
@@ -731,58 +733,33 @@ def get_response(self, offset, e_true, e_reco=None):
-
-        migra_e_reco=e_reco / e_true
-        integral=np.zeros(len(e_reco) - 1)
-        if norm != 0:
-            for i, migra_int in enumerate(migra_e_reco[:-1]):
-                # The migra_e_reco bin is inferior to the migra min in the dispersion fits file
-                if migra_e_reco[i + 1] < migra_mean[0]:
-                    continue
-                # The migra_e_reco bin is superior to the migra max in the dispersion fits file
-                elif migra_e_reco[i] > migra_mean[-1]:
-                    continue
-                else:
-                    if migra_e_reco[i] < migra_mean[0]:
-                        i_min=0
-                    else:
-                        i_min=np.where(migra_mean < migra_e_reco[i])[0][-1]
-                    i_max=np.where(migra_mean < migra_e_reco[i + 1])[0][-1]
-
-                    migra_bin_reco=migra_bin[i_min + 1:i_max]
-                    if migra_e_reco[i + 1] > migra_mean[-1]:
-                        index=np.arange(i_min, i_max, 1)
-                        migra_bin_reco=np.insert(migra_bin_reco, 0, (migra_mean[i_min + 1] - migra_e_reco[i]))
-                    elif migra_e_reco[i] < migra_mean[0]:
-                        if i_max == 0:
-                            index=np.arange(i_min, i_max + 1, 1)
-                        else:
-                            index=np.arange(i_min + 1, i_max + 1, 1)
-                        migra_bin_reco=np.append(migra_bin_reco, migra_e_reco[i + 1] - migra_mean[i_max])
-                    elif i_min == i_max:
-                        index=np.arange(i_min, i_max + 1, 1)
-                        migra_bin_reco=np.insert(migra_bin_reco, 0, (migra_e_reco[i + 1] - migra_e_reco[i]))
-                    else:
-                        index=np.arange(i_min, i_max + 1, 1)
-                        migra_bin_reco=np.insert(migra_bin_reco, 0, (migra_mean[i_min + 1] - migra_e_reco[i]))
-                        migra_bin_reco=np.append(migra_bin_reco, migra_e_reco[i + 1] - migra_mean[i_max])
-                    val=self.data.evaluate(offset=offset, e_true=e_true, migra=migra_mean[index])
-                    integral[i]=np.sum(val * migra_bin_reco) / norm
+        # migration value of e_reco bounds
+        migra_e_reco = e_reco / e_true
+
+        # Define a vector of migration with mig_step step
+        mrec_min = self.migra.lo[0]
+        mrec_max = self.migra.hi[-1]
+        mig_array = np.arange(mrec_min, mrec_max, mig_step)
+
+        # Compute energy dispersion probabilty dP/dm for each element of migration array
+        vals = self.data.evaluate(offset=offset, e_true=e_true, migra=mig_array)
+
+        # Compute normalized cumulative sum to prepare integration
+        tmp = np.cumsum(vals) / np.sum(vals)
+
+        # Determine positions (bin indices) of e_reco bounds in migration array
+        # We ensure that no negative values are found
+        pos_mig = np.maximum(np.digitize(migra_e_reco, mig_array)-1 , 0)
```

Comment:
This line is quite involved I think. Could you split it up in 2? Like you already did with the comments

```
# Determine positions (bin indices) of e_reco bounds in migration array
pos_mig_temp = np.digitize(migra_e_reco, mig_array)-1

# We ensure that no negative values are found
pos_mig = np.maximum(pos_mig_temp , 0)
```